<launch>

    <!-- Argument to select the label space -->
    <!-- Use 'ade20k' to filter based on the yaml file, or any other value to process all objects -->
    <arg name="label_space" default="ade20k" />
    
    <!-- Topic arguments for flexible configuration -->
    <arg name="rgb_topic" default="/hydra_ros_node/input/left_cam/rgb/image_raw" />
    <arg name="camera_info_topic" default="/hydra_ros_node/input/left_cam/rgb/camera_info" />
    <arg name="mesh_topic" default="/hydra_dsg_visualizer/dsg_mesh" />
    <arg name="debug_save_images" default="false" />
    <!-- Start the Python service for physical property inference -->
    <node name="phy_graph_server"
          pkg="phy_graph"
          type="module.py"
          output="screen">
    </node>

    <!-- Start the C++ node for subscribing to Hydra's output -->
    <node name="phy_graph_node"
          pkg="phy_graph"
          type="phy_graph_node"
          output="screen">
        <!-- Remap topics to use arguments for flexibility -->
        <remap from="input_mesh" to="$(arg mesh_topic)" />
        <remap from="rgb_image" to="$(arg rgb_topic)" />
        <remap from="camera_info" to="$(arg camera_info_topic)" />

        <!-- Load parameters into the node's private namespace -->
        <rosparam command="load" file="$(find phy_graph)/config/ade20k.yaml" />
        <param name="label_space" value="$(arg label_space)" />
        <param name="/debug_save_images" value="$(arg debug_save_images)"/>
    </node>
</launch>
